src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala

```diff
@@ -63,9 +62,21 @@ class FFIDMetadataService(ffidMetadataRepository: FFIDMetadataRepository, matche
     }
   }
 
-  private def checkForInvalidPuids(matchesMetadata: FFIDMetadataInput): String = {
-    matchesMetadata.matches.map(_.puid.getOrElse("")) match {
-      case puids if puids.exists(passwordProtectedPuids.contains) => PasswordProtected
+  private def generateFileStatusRows(ffidMetadata: FFIDMetadataInput): List[FilestatusRow] = {
+    val uniqueStatuses: List[String] = ffidMetadata.matches.map(m => checkStatus(m.puid)).distinct
+
+    uniqueStatuses match {
+      case s if uniqueStatuses.size == 1 =>
```

Comment:
This `s` variable is unused. Can you use it in the line 70 instead of using uniqueStatuses?